### PR TITLE
Add support for Gemfile and enable molecule tests

### PR DESCRIPTION
**Pull request by sbesson**
This PR improves test coverages and adds features to account for various types of Jekyll repository:

- sets up `molecule` to test this role
- adds support for top-level `Gemfile` files in a Jekyll repository:
  - installs `nokogiri` dependencies
  - adds a `bundle update` handler to install/update Ruby dependencies if a `Gemfile` is present
- attempts to fix the deployment issue raised in https://github.com/IDR/idr.openmicroscopy.org/pull/22 by removing curly braces in conditional (see https://github.com/ansible/ansible/issues/13270).
- migrates the `jekyll build` step as a handler rather than a conditional task
- adds a `debug` statement to preserve the backwards-compatibility of the `jekyll_build_force_rebuild` option and deprecate as handlers can be more generically forced using `--force-handlers`

### CODE REVIEW

**Comment by sbesson** [comment]
With the last commit

```
[ANSIBLE0016] Tasks that run when changed should likely be handlers
/home/travis/build/openmicroscopy/ansible-role-jekyll-build/tasks/main.yml:61
Task/Handler: jekyll | update bundle
[ANSIBLE0016] Tasks that run when changed should likely be handlers
/home/travis/build/openmicroscopy/ansible-role-jekyll-build/tasks/main.yml:72
Task/Handler: jekyll | build jekyll website
```

is kind of unfortunate for the progress on this effort but nicely flags pieces of this code where refactoring is relevant. I might look into the migration of the `bundle install` and `jekyll build` steps as role handlers assuming no-one has an better idea or immediate need for this work. /cc @joshmoore

File: .travis.yml [hunk 1]
```diff
@@ -1,29 +1,15 @@
...
-# Install ansible
-addons:
-  apt:
-    packages:
-    - python-pip
 
 install:
-  # Install ansible
-  - pip install ansible
+- pip install ansible docker 'molecule<1.22'
```
Comment (joshmoore) [inline .travis.yml]: `ome-ansible-molecule-dependencies`?

File: molecule.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        cpus: 1
+  instances:
+    - name: jekyll-build
+
+docker:
+  containers:
+  - name: jekyll-build
+    image: centos/systemd
+    image_version: latest
+    privileged: True
```
Comment (manics) [inline molecule.yml]: Does the plain centos image work?
```
  image: centos
  image_version: 7
```

Comment (sbesson) [reply molecule.yml]: Yes, thanks for spotting the copy-and-paste of another roles (depending on `systemd` most likely) - pushed as 204ba1b.

**Comment by sbesson** [comment]
@manics what do you think of the status/todos on this PR? If that seems like an overall improvement, we could get this merged, tagged and deployed `1.1.0`. Then the IDR playbooks could try to upgrade to this version to see if it solves some of the issues with updating existing deployment. /cc @joshmoore

File: molecule.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+docker:
+  containers:
+  - name: jekyll-build
+    image: centos
+    image_version: 7
+
+ansible:
+  host_vars:
+    jekyll-build:
```
Comment (manics) [inline molecule.yml]: Could you move this var into the playbook for clarity? host-vars are useful when you're testing multiple hosts with different configs, but in this case there's only one.

File: tests/test_default.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import testinfra.utils.ansible_runner
```
Comment (manics) [inline tests/test_default.yml]: This file should be called `test_default.py`

File: tests/test_default.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+testinfra_hosts = testinfra.utils.ansible_runner.AnsibleRunner(
+    '.molecule/ansible_inventory').get_hosts('all')
+
+
+def test_hosts_file(File):
```
Comment (manics) [inline tests/test_default.yml]: Could you change this to test for the existence of a file generated by jekyll, such as `/var/www/localhost/html/index.html`. Optionally grep for the existence of a templated string that should be substituted in the file after generation. Example test syntax: https://github.com/openmicroscopy/ansible-role-postgresql/blob/2.0.0/tests/test_postgresql_94.py#L23

**Comment by manics** [comment]
Looks generally fine, though I haven't had a chance to test updates of existing repos yet.

One issue for others to consider (pre-existing issue so not a blocker): The default owner is `root` which means the default location is [`/home/root`](https://github.com/openmicroscopy/ansible-role-jekyll-build/blob/master/defaults/main.yml#L7). Would [`/srv`](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard#Directory_structure) be better?

**Review by manics — CHANGES_REQUESTED** [review]

**Comment by sbesson** [comment]
Pushed all changes to the test as suggested in https://github.com/openmicroscopy/ansible-role-jekyll-build/pull/1#pullrequestreview-43220449. Re default location, probably worth revisiting in a follow-up PR.

**Comment by manics** [comment]
LGTM

**Comment by joshmoore** [comment]
Targeted tag? 2.0.0?

**Comment by manics** [comment]
Depends whether you want the location to be changed or not.

**Comment by sbesson** [comment]
As mentioned in https://github.com/openmicroscopy/ansible-role-jekyll-build/pull/1#issuecomment-307100746, I would be fine with a `1.1.0`. My expectation is that this PR is adding features in a backwards-compatible manner

**Comment by joshmoore** [comment]
@manics?

**Comment by manics** [comment]
Sounds fine to me.

**Pull request merged**